chain/client/src/stateless_validation/partial_witness/partial_witness_actor.rs

```diff
@@ -244,13 +241,30 @@ impl PartialWitnessActor {
         Ok(())
     }
 
-    /// Handles the state witness ack message from the chunk validator.
-    /// It computes the round-trip time between sending the state witness and receiving
-    /// the ack message and updates the corresponding metric with it.
-    /// Currently we do not raise an error for handling of witness-ack messages,
-    /// as it is used only for tracking some networking metrics.
-    pub fn handle_chunk_state_witness_ack(&mut self, witness_ack: ChunkStateWitnessAck) {
-        self.state_witness_tracker.on_witness_ack_received(witness_ack);
+    /// Sends the witness part to the chunk validators, except for the following:
+    /// 1) The current validator, 2) Chunk producer that originally generated the witness part.
+    fn forward_state_witness_part(
+        &self,
+        partial_witness: PartialEncodedStateWitness,
+        chunk_validators: Vec<AccountId>,
```

Comment:
For safely, we can generate the chunk_validator vec within this function instead of passing it in as param?

Reply:
Yeah since the list of chunk validators were provided for `send_state_witness_parts` already, I decided to just pass it there. But now given your other comment, it would not be terrible to just compute the validators again in `forward_state_witness_part` to avoid any potential problems with that list being modified.

Reply:
While originally writing the code, I remember caching the chunk validator computation in epoch manager. It should be cheap to call it multiple times from epoch manager.